src/components/MonthlyChecking/MenteeCheckInFormModal.component.tsx

```diff
@@ -0,0 +1,237 @@
+  };
+
+  const handleRemoveLink = (index: number) => {
+    const newLinks = mediaLinks.filter((_, i) => i !== index);
+    setMediaLinks(newLinks);
+    setValue('mediaContentLinks', newLinks);
+  };
+
+  const handleLinkChange = (index: number, value: string) => {
+    const newLinks = mediaLinks.map((link, i) => (i === index ? value : link));
+    setMediaLinks(newLinks);
+    setValue('mediaContentLinks', newLinks);
+  };
+
+  const { submitCheckIn } = useSubmitCheckIn();
+  const [loading, setLoading] = useState(false);
+
+  const onSubmit = async (data: MenteeCheckInForm) => {
+    const checkInData = {
+      ...data,
+      mediaLinks: mediaLinks.filter((link) => link.trim() !== ''),
+      menteeId,
+    };
+    setLoading(true);
+    try {
+      await submitCheckIn(checkInData);
+      reset();
+      onClose();
+    } catch (error) {
+      console.error('Error submitting check-in:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  if (!isOpen) return null;
+
+  return (
+    <>
+      <div className="fixed inset-0 flex items-center justify-center z-50 bg-black bg-opacity-50">
+        <div className="bg-white rounded-lg shadow-lg w-11/12 md:w-1/3 max-h-full overflow-y-auto p-6">
+          <div className="flex justify-between items-center mb-4">
+            <h2 className="text-xl font-semibold">Monthly Check-In</h2>
+            <button onClick={onClose} aria-label="Close">
+              <img src={closeIcon} alt="Close" className="w-6 h-6" />
+            </button>
+          </div>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <div className="mb-4">
+              <label className="block text-sm font-medium text-gray-700">
+                Month
+              </label>
+              <p className="text-xs text-gray-500">
+                Select the month for which you are submitting the progress.
+              </p>
+              <select
+                {...register('title', { required: 'Month is required' })}
+                className={`mt-1 block w-full border ${
+                  errors.title ? 'border-red-500' : 'border-gray-300'
+                } rounded-md shadow-sm focus:ring-blue-500 focus:border-blue-500`}
+              >
+                <option value="">Select a month</option>
+                <option value="January">January</option>
+                <option value="February">February</option>
+                <option value="March">March</option>
+                <option value="April">April</option>
+                <option value="May">May</option>
+                <option value="June">June</option>
+                <option value="July">July</option>
+                <option value="August">August</option>
+                <option value="September">September</option>
+                <option value="October">October</option>
+                <option value="November">November</option>
+                <option value="December">December</option>
```

Comment:
Create a loop for this

const months = ['January', 'February', 'March' ....]

months.map()